src/sentry/web/api.py

```diff
@@ -587,6 +587,53 @@ def process(self, request, project, key, auth, helper, data, attachments=None, *
                              key, remote_addr, helper, attachments)
 
 
+class EventAttachmentStoreView(StoreView):
+
+    def post(self, request, project, event_id, **kwargs):
+        if not features.has('organizations:event-attachments',
+                            project.organization, actor=request.user):
+            raise APIForbidden("Event attachments are not enabled for this organization.")
+
+        if len(request.FILES) == 0:
+            return HttpResponse(status=400)
+
+        for name, uploaded_file in six.iteritems(request.FILES):
+            file = File.objects.create(
+                name=uploaded_file.name,
+                type='event.attachment',
+                headers={'Content-Type': uploaded_file.content_type},
+            )
+            file.putfile(uploaded_file)
+
+            # To avoid a race with EventManager which tries to set the group_id on attachments received before
+            # the event, first insert the attachment, then lookup for the event for its group.
+            event_attachment = EventAttachment.objects.create(
+                project_id=project.id,
+                event_id=event_id,
+                name=uploaded_file.name,
+                file=file,
+            )
+
+            try:
+                event = Event.objects.get(
```

Comment:
@mattrobenolt I heard there are plans to make changes to this API. Do you recommend doing this differently?